include/sunadaptcontroller/sunadaptcontroller_mrihtol.h

```diff
@@ -0,0 +1,88 @@
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the header file for the SUNAdaptController_MRIHTol module.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNADAPTCONTROLLER_MRIHTOL_H
+#define _SUNADAPTCONTROLLER_MRIHTOL_H
+
+#include <stdio.h>
+#include <sundials/sundials_adaptcontroller.h>
+
+#ifdef __cplusplus /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/* --------------------------------------------
+ * MRI H+tolerance implementation of SUNAdaptController
+ * -------------------------------------------- */
+
+struct SUNAdaptControllerContent_MRIHTol_
+{
+  SUNAdaptController HControl;
+  SUNAdaptController TolControl;
+  sunrealtype inner_max_relch;
+  sunrealtype inner_min_tolfac;
+  sunrealtype inner_max_tolfac;
+};
+
+typedef struct SUNAdaptControllerContent_MRIHTol_* SUNAdaptControllerContent_MRIHTol;
+
+/* ------------------
+ * Exported Functions
+ * ------------------ */
+
+SUNDIALS_EXPORT
+SUNAdaptController SUNAdaptController_MRIHTol(SUNContext sunctx,
+                                              SUNAdaptController HControl,
+                                              SUNAdaptController TolControl);
+SUNDIALS_EXPORT
+SUNErrCode SUNAdaptController_SetParams_MRIHTol(SUNAdaptController C,
+                                                sunrealtype inner_max_relch,
+                                                sunrealtype inner_min_tolfac,
+                                                sunrealtype inner_max_tolfac);
+SUNDIALS_EXPORT
+SUNAdaptController SUNAdaptController_GetSlowController_MRIHTol(
+  SUNAdaptController C);
+SUNDIALS_EXPORT
+SUNAdaptController SUNAdaptController_GetFastController_MRIHTol(
+  SUNAdaptController C);
```

Comment:
I agree that this is mixed throughout SUNDIALS, with some getters using this approach (e.g., retrieving subvectors from a ManyVector, or array pointers from a vector; or for that matter, all of our integrator constructors), while others use the latter (e.g., getting the current Butcher tables from ARKStep and ERKStep).

FWIW, I prefer this style, particularly since we can easily check to see whether the returned object is `NULL`.